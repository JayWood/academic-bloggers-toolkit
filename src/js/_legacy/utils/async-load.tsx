// TODO: deprecate this file
import React from 'react';

interface Props {
    component: Promise<React.ComponentClass<any>>;
    [k: string]: any;
}

interface State {
    component: React.ComponentClass<any> | null;
}

export default class AsyncLoad extends React.PureComponent<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            component: null,
        };
        this.onResolve();
    }

    render(): JSX.Element | null {
        const Component = this.state.component;
        const { component, ...props } = this.props;
        return Component === null ? null : <Component {...props} />;
    }

    private async onResolve(): Promise<void> {
        this.props.component.then(component => {
            this.setState(prevState => ({ ...prevState, component }));
        });
    }
}
