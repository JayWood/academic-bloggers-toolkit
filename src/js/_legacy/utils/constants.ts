export enum MenuActionType {
    CHANGE_STYLE = 'CHANGE_STYLE',
    DESTROY_PROCESSOR = 'DESTROY_PROCESSOR',
    INSERT_STATIC_BIBLIOGRAPHY = 'INSERT_STATIC_BIBLIOGRAPHY',
    OPEN_IMPORT_DIALOG = 'OPEN_IMPORT_DIALOG',
    REFRESH_PROCESSOR = 'REFRESH_PROCESSOR',
}

interface LocaleMapper {
    [k: string]: string | undefined;
}

/**
 * This object converts the locale names in wordpress or PubMed (keys) to the locales
 * in CSL (values). If CSL doesn't have a locale for a given WordPress locale,
 * then undefined is used (which will default to en-US)
 */
export const localeMapper: LocaleMapper = {
    af: 'af-ZA',
    'af-ZA': 'af-ZA',
    afr: 'af-ZA',
    ar: 'ar',
    'ar-AR': 'ar',
    ara: 'ar',
    arq: 'ar',
    ary: 'ar',
    az: 'tr-TR',
    az_TR: 'tr-TR',
    'bg-BG': 'bg-BG',
    bg_BG: 'bg-BG',
    bul: 'bg-BG',
    ca: 'ca-AD',
    'ca-AD': 'ca-AD',
    cat: 'ca-AD',
    chi: 'zh-CN',
    'cs-CZ': 'cs-CZ',
    cs_CZ: 'cs-CZ',
    cy: 'cy-GB',
    cze: 'cs-CZ',
    'da-DK': 'da-DK',
    da_DK: 'da-DK',
    dan: 'da-DK',
    'de-AT': 'de-AT',
    'de-CH': 'de-CH',
    'de-DE': 'de-DE',
    de_CH: 'de-CH',
    de_DE: 'de-DE',
    dut: 'nl-NL',
    el: 'el-GR',
    'el-GR': 'el-GR',
    'en-GB': 'en-GB',
    'en-US': 'en-US',
    en_GB: 'en-GB',
    'es-AR': 'es-ES',
    'es-CL': 'es-CL',
    'es-ES': 'es-ES',
    'es-MX': 'es-MX',
    es_AR: 'es-ES',
    es_CL: 'es-CL',
    es_CO: 'es-CL',
    es_ES: 'es-ES',
    es_GT: 'es-ES',
    es_MX: 'es-MX',
    es_PE: 'es-CL',
    es_PR: 'es-CL',
    es_VE: 'es-CL',
    est: 'et-ET',
    et: 'et-ET',
    'et-ET': 'et-ET',
    eu: 'eu',
    'fa-IR': 'fa-IR',
    fa_AF: 'fa-IR',
    fa_IR: 'fa-IR',
    fi: 'fi-FI',
    'fi-FI': 'fi-FI',
    fin: 'fi-FI',
    'fr-CA': 'fr-CA',
    'fr-FR': 'fr-FR',
    fr_BE: 'fr-FR',
    fr_CA: 'fr-CA',
    fr_FR: 'fr-FR',
    fre: 'fr-FR',
    ger: 'de-DE',
    gre: 'el-GR',
    gsw: 'de-CH',
    'he-IL': 'he-IL',
    he_IL: 'he-IL',
    heb: 'he-IL',
    hr: 'hr-HR',
    'hr-HR': 'hr-HR',
    hrv: 'hr-HR',
    'hu-HU': 'hu-HU',
    hu_HU: 'hu-HU',
    hun: 'hu-HU',
    ice: 'is-IS',
    'id-ID': 'id-ID',
    id_ID: 'id-ID',
    ind: 'id-ID',
    'is-IS': 'is-IS',
    is_IS: 'is-IS',
    'it-IT': 'it-IT',
    it_IT: 'it-IT',
    ita: 'it-IT',
    ja: 'ja-JP',
    'ja-JP': 'ja-JP',
    jpn: 'ja-JP',
    km: 'km-KH',
    'km-KH': 'km-KH',
    'ko-KR': 'ko-KR',
    ko_KR: 'ko-KR',
    kor: 'ko-KR',
    lav: 'lv-LV',
    lb_LU: 'lt-LT',
    lit: 'lt-LT',
    'lt-LT': 'lt-LT',
    lt_LT: 'lt-LT',
    lv: 'lv-LV',
    'lv-LV': 'lv-LV',
    mn: 'mn-MN',
    'mn-MN': 'mn-MN',
    'nb-NO': 'nb-NO',
    nb_NO: 'nb-NO',
    nl: 'nl-NL',
    'nl-NL': 'nl-NL',
    nl_BE: 'nl-NL',
    nl_NL: 'nl-NL',
    'nn-NO': 'nn-NO',
    nn_NO: 'nn-NO',
    nor: 'nn-NO',
    per: 'fa-IR',
    'pl-PL': 'pl-PL',
    pl_PL: 'pl-PL',
    pol: 'pl-PL',
    por: 'pt-PT',
    'pt-BR': 'pt-BR',
    'pt-PT': 'pt-PT',
    pt_BR: 'pt-BR',
    pt_PT: 'pt-PT',
    'ro-RO': 'ro-RO',
    ro_RO: 'ro-RO',
    'ru-RU': 'ru-RU',
    ru_RU: 'ru-RU',
    rum: 'ro-RO',
    rus: 'ru-RU',
    'sk-SK': 'sk-SK',
    sk_SK: 'sk-SK',
    'sl-SI': 'sl-SI',
    sl_SI: 'sl-SI',
    slo: 'sk-SK',
    slv: 'sl-SI',
    spa: 'es-ES',
    'sr-RS': 'sr-RS',
    sr_RS: 'sr-RS',
    srp: 'sr-RS',
    'sv-SE': 'sv-SE',
    sv_SE: 'sv-SE',
    swe: 'sv-SE',
    th: 'th-TH',
    'th-TH': 'th-TH',
    tha: 'th-TH',
    'tr-TR': 'tr-TR',
    tr_TR: 'tr-TR',
    tur: 'tr-TR',
    uk: 'uk-UA',
    'uk-UA': 'uk-UA',
    ukr: 'uk-UA',
    vi: 'vi-VN',
    'vi-VN': 'vi-VN',
    vie: 'vi-VN',
    'zh-CN': 'zh-CN',
    'zh-TW': 'zh-TW',
    zh_CN: 'zh-CN',
    zh_HK: 'zh-CN',
    zh_TW: 'zh-TW',
};
