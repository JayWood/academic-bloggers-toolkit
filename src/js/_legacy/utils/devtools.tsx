import AsyncLoad from '_legacy/utils/async-load';
import React from 'react';

export default function devtool(props: any): any {
    if (process.env.NODE_ENV !== 'production') {
        return (
            <AsyncLoad
                component={import('mobx-react-devtools').then(
                    mod => mod.default,
                )}
                {...props}
            />
        );
    }
    return (): null => null;
}

export function configureDevtool(options: {
    graphEnabled?: boolean;
    logEnabled?: boolean;
    updatesEnabled?: boolean;
    logFilter?(p: any): boolean;
}): void {
    if (process.env.NODE_ENV !== 'production') {
        const cdt = require('mobx-react-devtools').configureDevtool;
        cdt(options);
    }
}
