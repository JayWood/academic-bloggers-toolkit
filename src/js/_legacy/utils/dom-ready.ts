/**
 * A better jQuery `$(document).ready()`
 */
export default async function domReady(): Promise<{}> {
    return new Promise<{}>(
        (resolve): void => {
            if (document.readyState !== 'loading') {
                return resolve();
            } else {
                document.addEventListener('DOMContentLoaded', resolve);
            }
        },
    );
}
