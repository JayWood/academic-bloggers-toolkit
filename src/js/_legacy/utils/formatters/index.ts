export { formatBibliography } from './format-bibliography';
export { formatReferenceLinks } from './format-reference-links';
