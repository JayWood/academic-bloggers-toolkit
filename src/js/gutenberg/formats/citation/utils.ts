interface CCHProps {
    id: string;
    items: string[];
}

export function createCitationHtml({ id, items }: CCHProps): string {
    const TERMINATOR = '&#65279;';
    const citation = document.createElement('span');
    citation.className = 'abt-citation';
    citation.dataset.id = id;
    citation.dataset.items = JSON.stringify(items);
    citation.contentEditable = 'false';
    return `${citation.outerHTML}${TERMINATOR}`;
}
