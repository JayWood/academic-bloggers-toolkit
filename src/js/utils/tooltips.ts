type Position = 'top' | 'right' | 'bottom' | 'left';

/**
 * Utility function used to generate and append tooltips.
 * @param target Target element
 * @param tip Tooltip text
 * @param position Position of tooltip
 */
export function createTooltip(target: HTMLElement, tip: string, position: Position): void {
    const rect = target.getBoundingClientRect();
    const top = rect.top + rect.height / 2;
    let left = rect.left + rect.width / 2;

    const el = document.createElement('div');
    el.id = 'abt-tooltip';
    el.innerText = tip;
    el.classList.add('abt-tooltip');
    document.body.appendChild(el);

    const marginLeft = -1 * (el.offsetWidth / 2) || 0;
    const marginTop = -1 * (el.offsetHeight / 2) || 0;

    if (position === 'left' || position === 'right') {
        left = rect.width / 2;
        if (top + marginTop < 0) {
            el.style.top = '0';
            el.style.marginTop = '0';
        } else {
            el.style.top = top + 'px';
            el.style.marginTop = marginTop + 'px';
        }
    } else {
        if (left + marginLeft < 0) {
            el.style.left = '0';
            el.style.marginLeft = '0';
        } else {
            el.style.left = left + 'px';
            el.style.marginLeft = marginLeft + 'px';
        }
    }

    switch (position) {
        case 'top':
            el.style.top = rect.top - el.offsetHeight - 10 + 'px';
            break;
        case 'right':
            el.style.left = rect.left + rect.width + 10 + 'px';
            break;
        case 'bottom':
            el.style.top = rect.top + rect.height + 10 + 'px';
            break;
        case 'left':
        default:
            el.style.left = rect.left - el.offsetWidth - 10 + 'px';
    }

    el.classList.add('abt-tooltip_active');
}

/**
 * Utility function used to destroy existing tooltips
 */
export function destroyTooltip(): void {
    const el = document.getElementById('abt-tooltip');
    if (el) el.parentElement!.removeChild(el);
}
