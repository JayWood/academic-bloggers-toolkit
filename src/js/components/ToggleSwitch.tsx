import * as React from 'react';
import { createTooltip, destroyTooltip } from 'utils/Tooltips';

interface Props {
    checked: boolean;
    label: string;
    onChange(e: React.FormEvent<HTMLInputElement>): void;
}

export class ToggleSwitch extends React.PureComponent<Props, {}> {
    handleMouseOver = (e: React.MouseEvent<HTMLLabelElement>) => {
        createTooltip(e.currentTarget, e.currentTarget.getAttribute('data-tooltip')!, 'left');
    };

    render() {
        const { checked, onChange, label } = this.props;
        return (
            <div>
                <input
                    type="checkbox"
                    id="inline-toggle"
                    style={{ display: 'none' }}
                    checked={checked}
                    aria-checked={checked}
                    onChange={onChange}
                />
                <label
                    htmlFor="inline-toggle"
                    role="tooltip"
                    data-tooltip={label}
                    onMouseOver={this.handleMouseOver}
                    onMouseOut={destroyTooltip}
                />
                <style jsx>{`
                    label {
                        position: relative;
                        display: block;
                        margin: 0 10px;
                        height: 20px;
                        width: 44px;
                        background: #898989;
                        border-radius: 100px;
                        cursor: pointer;
                        -webkit-transition: all 0.3s ease;
                        transition: all 0.3s ease;
                    }
                    label:after {
                        position: absolute;
                        left: -2px;
                        top: -3px;
                        display: block;
                        width: 26px;
                        height: 26px;
                        border-radius: 100px;
                        background: #fff;
                        -webkit-box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                        box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                        content: '';
                        -webkit-transition: all 0.3s ease;
                        transition: all 0.3s ease;
                    }
                    label:active:after {
                        -webkit-transform: scale(1.15, 0.85);
                        transform: scale(1.15, 0.85);
                    }
                    input:checked ~ label {
                        background: #6f9eb1;
                    }
                    input:checked ~ label:after {
                        left: 20px;
                        background: #0085ba;
                    }
                    input:disabled ~ label {
                        background: #d5d5d5;
                        pointer-events: none;
                    }
                    input:disabled ~ label:after {
                        background: #bcbdbc;
                    }
                `}</style>
            </div>
        );
    }
}
