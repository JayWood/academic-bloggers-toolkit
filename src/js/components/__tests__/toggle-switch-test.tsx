jest.mock('utils/tooltips');

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { observable } from 'mobx';
import * as React from 'react';

import { createTooltip, destroyTooltip } from 'utils/tooltips';
import ToggleSwitch from '../toggle-switch';

const changeFn = () => void 0;

describe('<ToggleSwitch />', () => {
    it('should match snapshots', () => {
        const checked = observable(false);
        let component = shallow(
            <ToggleSwitch checked={checked} label="Test 1" onChange={changeFn} />,
        );
        expect(toJSON(component)).toMatchSnapshot();

        checked.set(true);
        component = shallow(<ToggleSwitch checked={checked} label="Test 2" onChange={changeFn} />);
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should create tooltip on mouseover', () => {
        const component = shallow(
            <ToggleSwitch checked={observable(false)} label="Test 3" onChange={changeFn} />,
        );
        const label = component.find('label');
        expect(createTooltip).not.toHaveBeenCalled();
        expect(destroyTooltip).not.toHaveBeenCalled();
        label.simulate('mouseOver', { currentTarget: { dataset: { tooltip: 'test' } } });
        label.simulate('mouseOut');
        expect(createTooltip).toHaveBeenCalledTimes(1);
        expect(destroyTooltip).toHaveBeenCalledTimes(1);
    });
});
