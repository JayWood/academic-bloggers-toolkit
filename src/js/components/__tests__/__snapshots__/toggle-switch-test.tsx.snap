// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> should match snapshots 1`] = `
<div>
  <Tooltip
    active={false}
    id="inline-toggle"
    text="Test 1"
    transform=""
  />
  <input
    aria-checked={false}
    aria-describedby="inline-toggle"
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    aria-checked={false}
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="checkbox"
  />
  <style
    jsx={true}
  >
    
                        input {
                            opacity: 0 !important;
                            pointer-events: none;
                            position: absolute;
                            z-index: 1;
                        }
                        span {
                            position: relative;
                            display: block;
                            margin: 5px;
                            height: 20px;
                            width: 44px;
                            background: #898989;
                            border-radius: 100px;
                            cursor: pointer;
                            transition: all 0.3s ease;
                        }
                        span:after {
                            position: absolute;
                            left: -2px;
                            top: -3px;
                            display: block;
                            width: 26px;
                            height: 26px;
                            border-radius: 100px;
                            background: #fff;
                            box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            content: '';
                            transition: all 0.3s ease;
                        }
                        span:active:after {
                            -webkit-transform: scale(1.15, 0.85);
                            transform: scale(1.15, 0.85);
                        }
                        input:checked + span {
                            background: #6f9eb1;
                        }
                        input:checked + span:after {
                            left: 20px;
                            background: #0085ba;
                        }
                        input:disabled + span {
                            background: #d5d5d5;
                            pointer-events: none;
                        }
                        input:disabled + span:after {
                            background: #bcbdbc;
                        }
                        input:disabled:checked + span {
                            background: #6f9eb1;
                            opacity: 0.7;
                        }
                        input:disabled:checked + span:after {
                            background: #0085ba;
                            opacity: 0.7;
                        }
                        input:focus + span {
                            outline: 1.1px auto rgba(19, 124, 189, 0.5);
                            outline-width: 1.1px;
                            outline-style: solid;
                            outline-offset: 5px;
                            transition: none;
                        }
                    
  </style>
</div>
`;

exports[`<ToggleSwitch /> should match snapshots 2`] = `
<div>
  <Tooltip
    active={false}
    id="inline-toggle"
    text="Test 2"
    transform=""
  />
  <input
    aria-checked={true}
    aria-describedby="inline-toggle"
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    aria-checked={true}
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    role="checkbox"
  />
  <style
    jsx={true}
  >
    
                        input {
                            opacity: 0 !important;
                            pointer-events: none;
                            position: absolute;
                            z-index: 1;
                        }
                        span {
                            position: relative;
                            display: block;
                            margin: 5px;
                            height: 20px;
                            width: 44px;
                            background: #898989;
                            border-radius: 100px;
                            cursor: pointer;
                            transition: all 0.3s ease;
                        }
                        span:after {
                            position: absolute;
                            left: -2px;
                            top: -3px;
                            display: block;
                            width: 26px;
                            height: 26px;
                            border-radius: 100px;
                            background: #fff;
                            box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.05);
                            content: '';
                            transition: all 0.3s ease;
                        }
                        span:active:after {
                            -webkit-transform: scale(1.15, 0.85);
                            transform: scale(1.15, 0.85);
                        }
                        input:checked + span {
                            background: #6f9eb1;
                        }
                        input:checked + span:after {
                            left: 20px;
                            background: #0085ba;
                        }
                        input:disabled + span {
                            background: #d5d5d5;
                            pointer-events: none;
                        }
                        input:disabled + span:after {
                            background: #bcbdbc;
                        }
                        input:disabled:checked + span {
                            background: #6f9eb1;
                            opacity: 0.7;
                        }
                        input:disabled:checked + span:after {
                            background: #0085ba;
                            opacity: 0.7;
                        }
                        input:focus + span {
                            outline: 1.1px auto rgba(19, 124, 189, 0.5);
                            outline-width: 1.1px;
                            outline-style: solid;
                            outline-offset: 5px;
                            transition: none;
                        }
                    
  </style>
</div>
`;
